dspace-server-webapp/src/main/java/org/dspace/app/rest/hdlresolver/HdlResolverRestController.java

```diff
@@ -0,0 +1,112 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller is public and is useful for handle resolving,
+ * wheter a target handle identifier will be resolved into the
+ * corresponding URL (if found), otherwise will respond a null string.
+ * 
+ * @author Vincenzo Mecca (vins01-4science - vincenzo.mecca at 4science.it)
+ *
+ */
+@RestController
+@ConditionalOnProperty("handle.remote-resolver.enabled")
+@RequestMapping(path = "/{hdlService:hdlResolver|resolve|listhandles|listprefixes}/")
+public class HdlResolverRestController {
+    static final String HDL_RESOLVER = "/hdlResolver/";
+    static final String RESOLVE = "/resolve/";
+    static final String LISTHANDLES = "/listhandles/";
+    static final String LISTPREFIXES = "/listprefixes/";
+
+    private static final Logger log = LogManager.getLogger();
+
+    @Autowired
+    private HdlResolverService hdlResolverService;
+
+    /**
+     * REST GET Method used to find and retrieve the URL of a target Handle. It
+     * should return only one item, if found, else a null body value.
+     * </br>
+     * Generate an <code>HttpStatus.BAD_REQUEST</code> 400 Error if the handle used
+     * in path isn't valid.
+     * </br>
+     * The response type will be (<code>application/json;charset=UTF-8</code>)
+     * a string representing an array-like list of handlers:
+     * </br>
+     * Example:
+     *      <ul>
+     *          <li>Request: GET - http://{dspace.url}/hdlResolver/handleIdExample/1</li>
+     *          <li>Response: 200 - ["http://localhost/handle/hanldeIdExample1"]
+     *      </ul>
+     * 
+     * @param request HttpServletRequest
+     * @return One element List or <code>null</code> with status 200
+     *         <code>HttpStatus.OK</code> or <code>HttpStatus.BAD_REQUEST</code> 400
+     *         error
+     */
+    @GetMapping(
+        value = "**",
+        produces = "application/json;charset=UTF-8"
+    )
+    public ResponseEntity<String> getHandleUrlResolver(HttpServletRequest request, @PathVariable String hdlService) {
+        HdlResolverDTO handleResolver =
+               this.hdlResolverService.resolveBy(
+                       request.getRequestURI(),
+                       MessageFormat.format("{0}/{1}/", request.getContextPath(), hdlService)
+               );
+        if (!handleResolver.isValid()) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        } else {
+            return new ResponseEntity<>(this.resolveToURL(request, handleResolver), HttpStatus.OK);
+        }
+    }
+
```

Comment:
add here 
the implementation and mapping for the `<controller>/listprefixes`
https://github.com/DSpace/DSpace/blob/5e4aec74645ee0ef877952259347992d01353e84/dspace-jspui/src/main/java/org/dspace/app/webui/handle/HandleJSONResolver.java#L69-L83

and `<controller>/resolve/listhandles`
https://github.com/DSpace/DSpace/blob/5e4aec74645ee0ef877952259347992d01353e84/dspace-jspui/src/main/java/org/dspace/app/webui/handle/HandleJSONResolver.java#L85-L101